File: src/main/java/com/slytherin/slytherbyte/models/services/authentication/AuthenticationService.java
package com.slytherin.slytherbyte.models.services.authentication;

import com.slytherin.slytherbyte.models.exceptions.AuthenticationException;
import com.slytherin.slytherbyte.models.request.AuthenticationRequest;
import com.slytherin.slytherbyte.models.request.RegisterRequest;
import com.slytherin.slytherbyte.models.response.AuthenticationResponse;

public interface AuthenticationService {
    void register(RegisterRequest input) throws AuthenticationException;
    AuthenticationResponse login(AuthenticationRequest input);
}

File: src/app/services/authenticationService.ts
import { HttpClient } from "@angular/common/http";
import { inject, Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { RegisterRequest } from "../models/registerRequest";
import { AuthenticationRequest } from "../models/authenticationRequest";
import { AuthenticationResponse } from "../models/authenticationResponse";

@Injectable({ providedIn: 'root' })
export class AuthenticationService {
    private _url = 'http://localhost:8080/api/auth/register';
    private _http = inject(HttpClient);
    
    public sendRegisterRequest(obj: RegisterRequest): Observable<void> {
        return this._http.post<void>(this._url, obj);
    }

    public sendLoginRequest(obj: AuthenticationRequest): Observable<AuthenticationResponse> {
        return this._http.post<AuthenticationResponse>('http://localhost:8080/api/auth/login', obj);
    }
}